api/spec/spec.go

```diff
@@ -0,0 +1,234 @@
+package spec
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+// AssertHTTPAPISpec runs a test suite of HTTP requests against the given base URL
+// to assert it conforms to the qri core API specification. Spec is defined in
+// the "open_api_3.yaml" file in the api package
+func AssertHTTPAPISpec(t *testing.T, baseURL, specPackagePath string) {
+	t.Helper()
+
+	// Create a mock data server. Can't move this into the testRunner, because we need to
+	// ensure only this test is using the server's port "55555".
+	s := newMockDataServer(t)
+	defer s.Close()
+
+	base, err := url.Parse(baseURL)
+	if err != nil {
+		t.Fatalf("invalid base url: %s", err)
+	}
+
+	testFiles := []string{
+		"testdata/access.json",
+		"testdata/aggregate.json",
+		"testdata/automation.json",
+		"testdata/dataset.json",
+		"testdata/misc.json",
+		"testdata/peer.json",
+		"testdata/profile.json",
+		"testdata/remote.json",
+		"testdata/working_directory.json",
+
+		// sync.json is intentionally left out
+		// as it's more a protocol that doesn't belong
+		// in the RPC API
+		// "testdata/sync.json",
+	}
+
+	sl := mustLoadSkipList(t, filepath.Join(specPackagePath, "testdata/skip.json"))
+
+	for _, path := range testFiles {
+		t.Run(filepath.Base(path), func(t *testing.T) {
+			ts := mustLoadTestSuite(t, filepath.Join(specPackagePath, path))
+			for i, c := range ts {
+				skip := false
+				for _, s := range sl {
```

Comment:
How about extracting out the function `isInSkipList` so that this turns into a call `isInSkipList(sl, c.Endpoint)`?